### PR TITLE
Fix an Inlining bug with a name collision in a br nested in a call param

### PR DESCRIPTION
**Pull request by kripken**
This bug has existed for many years, but the fuzzer only found it now...

### CODE REVIEW

File: src/ir/branch-utils.h [hunk 1]
```diff
@@ -327,6 +327,12 @@ struct BranchAccumulator
     auto selfBranches = getUniqueTargets(curr);
     branches.insert(selfBranches.begin(), selfBranches.end());
   }
+
+  static NameSet get(Expression* tree) {
+    BranchAccumulator branches;
+    branches.walk(tree);
+    return branches.branches;
```
Comment (tlively) [inline src/ir/branch-utils.h:334]: I would find this naming clearer.
```suggestion
    BranchAccumulator accumulator;
    accumulator.walk(tree);
    return accumulator.branches;
```

File: src/passes/Inlining.cpp [hunk 2]
```diff
@@ -338,11 +338,25 @@ static Expression* doInlining(Module* module,
...
+  //      (call's first parameter
+  //        (br $X)         ;; nested br in call's first parameter
+  //      )
+  //    )
+  //
+  // (In this case we could use a second block and define the named block $X
+  // after the call's parameters, but that adds work for an extremely rare
+  // situation.)
+  if (BranchUtils::hasBranchTarget(from->body, block->name) ||
+      BranchUtils::BranchSeeker::has(call, block->name)) {
```
Comment (tlively) [inline src/passes/Inlining.cpp:354]: What is the difference between `BranchUtils::hasBranchTarget` and `BranchUtils::BranchSeeker::has`?

Comment (tlively) [reply src/passes/Inlining.cpp:354]: Got it, thanks. Does it save work to check for existence before gathering all the names below? It seems like gathering names would be about the same amount of work.

Comment (kripken) [reply src/passes/Inlining.cpp:354]: Gathering them requires us to build up a set (allocation+hashing) so it is a little more work at least. Probably not a lot in the general case. But it is extremely rare to have a collision here.

Comment (tlively) [reply src/passes/Inlining.cpp:354]: Makes sense 👍

**Pull request merged**